doc/dev/developer_guide/index.rst

```diff
@@ -777,7 +777,25 @@ Unit tests are declared in the ``CMakeLists.txt`` files (multiple files under
 which are themselves defined in ``./cmake/modules/AddCephTest.cmake``. Some
 unit tests are scripts, while others are binaries that are compiled during the
 build process.  The ``add_ceph_test`` function is used to declare unit test
-scripts, while ``add_ceph_unittest`` is used for unit test binaries.
+scripts, while ``add_ceph_unittest`` is used for unit test binaries. 
+``add_tox_test`` declare tests that require ``tox`` for testing and sets up 
+the full environment for this during testing.
+::
+
+        add_ceph_test(test_name test_path <args> <options>)
+        add_ceph_unittest(test_name <args> <options>)
+        add_tox_test(test_name <args> <options>)
+        args: any parameter, not being an options keyword
+        options:
+            [COST <float>]
+               Order of testing is set by COST. 
+               Highest COST are tested first.
```

Comment:
That's great! Thanks @wjwithagen. Just curious:
- Could you please briefly comment here where/how to obtain the cost you have already set in some tests.
- Could you please list the labels you already defined and give a brief explanation on each one and some suggestion/criteria for adding new labels (per-language?, per-kind of testing?, per-environment?...)?

Reply:
@epuertat
The remainder of the dev-doc is also rather tearse and assumes that `the source` is the definite truth. BUt I'll see if I can add some fine prose.

Reply:
Thanks a lot for doing it!